scripts/lis_hotfix_pipeline/Library.psm1

```diff
@@ -0,0 +1,235 @@
+            $NatRule = Get-AzureRmLoadBalancerInboundNatRuleConfig -LoadBalancer $LB2 -Name $NatRuleName  -ErrorAction SilentlyContinue
+            if ( $NatRule ) {
+                Write-LogInfo "Removing SSH Rule $NatRuleName ..."
+                [void]($LB2 | Remove-AzureRmLoadBalancerInboundNatRuleConfig -Name $NatRuleName -Verbose)
+                [void]($LB2 | Set-AzureRmLoadBalancer -Verbose )
+            }
+            Write-LogInfo "Checking NIC $NicName ..."
+            $NIC = Get-AzureRmNetworkInterface -ResourceGroupName $ResourceGroupName -Name $NicName -ErrorAction SilentlyContinue
+            if ( $NIC ) {
+                Write-LogInfo "Removing NIC $NicName ..."
+                $null = $NIC | Remove-AzureRmNetworkInterface -Force -Verbose -ErrorAction SilentlyContinue
+            }
+            Write-LogInfo "Clenaup Completed."
+            $retry = $false
+        } catch {
+            $maxRetry -= 1
+            $ErrorMessage = $_.Exception.Message
+            $ErrorLine = $_.InvocationInfo.ScriptLineNumber
+            Write-LogInfo "EXCEPTION : $ErrorMessage at line: $ErrorLine"
+            Write-LogInfo "Exception in the Start-FailureCleanup. Retrying... (Remaining attempts: $maxRetry)..."
+        }
+    }
+}
+
+Function Get-RPMPackageNames ($StorageContext, $FolderPath, $KernelVersion) {
+    $KernelURI = ""
+    $OtherPackages = ""
+    Write-LogInfo "Get-AzStorageBlob -Container kernel ..."
+    $RPMs = Get-AzureStorageBlob -Container kernel -Context $context
+    foreach ($rpm in $RPMs) {
+        $rpmURI = $($rpm.ICloudBlob.Uri.AbsoluteUri)
+        if ($rpmURI -imatch "/$KernelVersion/") {
+            if ($rpmURI -imatch "kernel-$KernelVersion.") {
+                $KernelURI = $rpmURI
+            } else {
+                $OtherPackages += "$rpmURI "
+            }
+        }
+    }
+    return $KernelURI,$OtherPackages
+}
+
+Function Install-KernelPackages ( $PublicIP, $SSHPort, $LinuxUsername, $LinuxPassword, $KernelPackage, $OtherPackages ) {
+    try {
+        $KernelBefore = Run-LinuxCmd -username $LinuxUsername -password $LinuxPassword -ip $PublicIP -port $SSHPort -command "uname -r"
+        $null = Run-LinuxCmd -username $LinuxUsername -password $LinuxPassword -ip $PublicIP -port $SSHPort -command "rpm -ivh $KernelPackage $OtherPackages"
+        $null = Run-LinuxCmd -username $LinuxUsername -password $LinuxPassword -ip $PublicIP -port $SSHPort -command "yum -y --nogpgcheck update gcc" -ignoreLinuxExitCode
+        $null = Run-LinuxCmd -username $LinuxUsername -password $LinuxPassword -ip $PublicIP -port $SSHPort -command "init 6" -ignoreLinuxExitCode
+        Write-LogInfo "Sleeping 30 seconds..."
+        Start-Sleep 30
+        $KernelAfter = Run-LinuxCmd -username $LinuxUsername -password $LinuxPassword -ip $PublicIP -port $SSHPort -command "uname -r" -maxRetryCount 20
+        Write-LogInfo "Kernel Before: $KernelBefore"
+        Write-LogInfo "Kernel After: $KernelAfter"
```

Comment:
debug print can be optimized by having single line comment

Reply:
Top-down alignment gives quick comparison.